util/codec/codec.go

```diff
@@ -359,13 +359,15 @@ func encodeHashChunkRowIdx(sc *stmtctx.StatementContext, row chunk.Row, tp *type
 			return
 		}
 	case mysql.TypeEnum:
-		flag = uvarintFlag
+		flag = compactBytesFlag
```

Comment:
`func encode(sc *stmtctx.StatementContext, b []byte, vals []types.Datum, comparable bool)` also has this problem, how about updating it together?

Reply:
That is used to encode storage value, change that would break compatibility.